File: src-server/modules/system/api/role.go
package api

import (
	"github.com/gin-gonic/gin"
	"github.com/zhouhp1295/g3-cms/boot"
	"github.com/zhouhp1295/g3-cms/modules/system/dao"
	"github.com/zhouhp1295/g3/net"
)

type _sysRoleApi struct {
	net.BaseApi
}

var SysRoleApi = &_sysRoleApi{
	net.BaseApi{Dao: dao.SysRoleDao},
}

const (
	PermSystemRoleList   = "system:role:list"
	PermSystemRoleQuery  = "system:role:query"
	PermSystemRoleAdd    = "system:role:add"
	PermSystemRoleEdit   = "system:role:edit"
	PermSystemRoleRemove = "system:role:remove"
)

func init() {
	boot.ApiGet("/admin/system/role/page", SysRoleApi.HandlePage, PermSystemRoleQuery)
	boot.ApiGet("/admin/system/role/get", SysRoleApi.HandleGet, PermSystemRoleQuery)
	boot.ApiPost("/admin/system/role/insert", SysRoleApi.HandleInsert, PermSystemRoleAdd)
	boot.ApiPut("/admin/system/role/update", SysRoleApi.HandleUpdate, PermSystemRoleEdit)
	boot.ApiPut("/admin/system/role/status", SysRoleApi.HandleUpdateStatus, PermSystemRoleEdit)
	boot.ApiDelete("/admin/system/role/delete", SysRoleApi.HandleDelete, PermSystemRoleRemove)
	//扩展接口
	boot.ApiGet("/admin/system/role/listOptions", SysRoleApi.listOptions)
}

func (api *_sysRoleApi) listOptions(ctx *gin.Context) {
	api.SuccessList(ctx, dao.SysRoleDao.RoleOptions())
}

File: src-backend/src/api/system/role.js
import request from '@/utils/request'

// 查询角色列表
export function listRoleOptions() {
  return request({
    url: '/api/admin/system/role/listOptions',
    method: 'get',
  })
}

// 查询角色列表
export function listRole(query) {
  return request({
    url: '/api/admin/system/role/page',
    method: 'get',
    params: query
  })
}

// 查询角色详细
export function getRole(roleId) {
  return request({
    url: '/api/admin/system/role/get',
    method: 'get',
    params: {id : roleId}
  })
}

// 新增角色
export function addRole(data) {
  return request({
    url: '/api/admin/system/role/insert',
    method: 'post',
    data: data
  })
}

// 修改角色
export function updateRole(data) {
  return request({
    url: '/api/admin/system/role/update',
    method: 'put',
    data: data
  })
}

// 角色状态修改
export function changeRoleStatus(id, status) {
  const data = {
    id,
    status
  }
  return request({
    url: '/api/admin/system/role/status',
    method: 'put',
    data: data
  })
}

// 删除角色
export function delRole(roleId) {
  return request({
    url: '/api/admin/system/role/delete',
    method: 'delete',
    params: {id : roleId}
  })
}
